ekglib/ldap_parser/parser.py

```diff
@@ -500,23 +506,27 @@ def _parse_attribute(self, key, values):
         elif key == 'entryUUID':
             self._parse_entry_uuid(values)
         elif key == 'jpegPhoto':
-
-            log(f"found jpegPhoto {values}")
-            try:
-                for v in values:
-                    b = bytes(v, 'utf-8')
-                    base64_bytes = base64.b64encode(b)
-                    log(f"found jpegPhoto {base64_bytes}")
-                    base64_url = "data:image/jpeg;charset=utf-8;base64," + base64_bytes
-                    log(f"found jpegPhoto {base64_url}")
-                    self._add((self.entry_iri, LDAP.term(key), base64_url))
-            except:
-                log_error("problem reading jpegPhoto")
-            return
+            self.parse_binary_content(key, values)
         else:
             for value in values:
                 self._add((self.entry_iri, LDAP.term(key), Literal(value)))
 
+    def parse_binary_content(self, key, values):
+        """Binary content is transformed into a Literal with base64 encoding"""
+        for value in values:
+            base64_bytes = b64encode(str_to_binary(value)).decode('ascii')
+
+            # @LAP: assume this is handy in the UI?
+            base64_url = "data:image/jpeg;charset=utf-8;base64," + base64_bytes
+            self._add((self.entry_iri, LDAP.term(key + "Url"), URIRef(base64_url)))
+
+            # NB: Can't use rdflib.Literal with the correct datatype xsd:base64Binary,
+            # as it messes with the value (decodes it), using a placeholder xsd:base64BinaryString
+            # which gets replaced in the output stream, see _substitute_iris
```

Comment:
does this imply that we should be able to give rdflib the Literal(binary) and have it base64encode?

Reply:
Yes, that would be ideal, but it actually encodes the python unicode, which is not the same as the jpeg image octet stream